acceptance/tests/options/config_file/ttls_cache_system_uptime.rb

```diff
@@ -56,39 +68,51 @@
     #   assert_equal(expected, cache_content)
     # end
 
-    step "calling facter without a query will cache the entire group" do
+    step 'calling facter without a query will cache the entire group' do
       _output = on(agent, facter)
 
       stdout = agent.cat("#{cache_folder}/uptime")
       cache_content = JSON.parse(stdout)
-      ["system_uptime.days",
-        "uptime_days",
-        "system_uptime.hours",
-        "uptime_hours",
-        "system_uptime.seconds",
-        "uptime_seconds",
-        "system_uptime.uptime",
-        "uptime",
-        "cache_format_version"].each do |key|
+      ['system_uptime.days',
+        'uptime_days',
+        'system_uptime.hours',
+        'uptime_hours',
+        'system_uptime.seconds',
+        'uptime_seconds',
+        'system_uptime.uptime',
+        'uptime',
+        'cache_format_version'].each do |key|
 
           assert_equal(true, cache_content.has_key?(key))
         end
     end
 
-    step "calling a single fact fron the cached group will not overwrite the file" do
+    operating_system = fact_on(agent, 'operatingsystem')
+
+    if operating_system == 'Ubuntu'
+      step 'check file info to verify that it did not change' do
+        output1 = on(agent, "stat -c %Y #{cache_folder}/uptime")
+        on(agent, facter)
+        on(agent, facter('system_uptime'))
+        output2 = on(agent, "stat -c %Y #{cache_folder}/uptime")
+        assert_equal(output1.stdout, output2.stdout)
+      end
+    end
+
+    step 'calling a single fact fron the cached group will not overwrite the file' do
```

Comment:
I think this step is not necessary because you make make sure the file is not overwritten, in the previous step. If you want, you can perform the previous step two times, requesting two different facts .